hbase-server/src/main/java/org/apache/hadoop/hbase/master/janitor/CatalogJanitor.java

```diff
@@ -317,11 +329,19 @@ static boolean cleanParent(MasterServices services, RegionInfo parent, Result ro
       LOG.debug("Deleting region " + parent.getShortNameToLog() + " because daughters -- " +
         daughterA + ", " + daughterB + " -- no longer hold references");
       ProcedureExecutor<MasterProcedureEnv> pe = services.getMasterProcedureExecutor();
-      pe.submitProcedure(new GCRegionProcedure(pe.getEnvironment(), parent));
-      // Remove from in-memory states
```

Comment:
GCRegionProcedure does this. It is not needed here too. I think it better to let the procedure do it. We expect the procedure to take care of such details, so let's in fact do that.
Keeping this code around would be harmless, though.